core/rx-java-driver/src/main/java/org/hawkular/rx/cassandra/driver/RxSessionImpl.java

```diff
@@ -38,9 +43,14 @@
 public class RxSessionImpl implements RxSession {
 
     private Session session;
+    private int maximumInflightRequests;
 
     public RxSessionImpl(Session session) {
         this.session = session;
+        maximumInflightRequests = Math.min(
```

Comment:
We don't know what host is being targeted in this case and I'm not sure if there's a cheap way to find out either (we would have to maintain a list of TokenRanges per host and calculate hash for each query and then check which hosts that might match)